packages/material-ui/src/Tooltip/Tooltip.js

```diff
@@ -407,6 +412,7 @@ const Tooltip = React.forwardRef(function Tooltip(props, ref) {
     document.addEventListener('keydown', handleKeyDown);
 
     return () => {
+      document.body.style.WebkitUserSelect = prevUserSelect.current;
```

Comment:
This should only be removed on unmount. This effect cleanup is unrelated.

Reply:
I disagree, the cleanup should be run as soon as the tooltip is closed no unmounted, as the keyboard handler.

Reply:
Actually, it could also be better to remove it at the end of the transition, so end-users can still select the text inside the tooltip.

Reply:
This cleanup is just a fallback. The correct thing to do would be to cleanup if the timer is cancelled.